game.ts

```diff
@@ -0,0 +1,29 @@
+import * as readline from "node:readline";
+import { stdin as input, stdout as output } from "node:process";
+
+//createInterface is a callable of Node's readLine module (see documentation)
+//In this way, we can connect stdin and stdout
+const rl = readline.createInterface({ input, output });
+
+function generateComputerMove() {
+  return String(Math.round(Math.random() * 2));
+}
+
+export function play(question: string) {
+  let computerMove = generateComputerMove();
+  rl.question(question, (userMove) => {
```

Comment:
In general I think promise-based APIs used with `await` can be more readable and also make it simpler to do correct error handling. So I'd suggest trying the promises API (`import * as readline from 'node:readline/promises';`). I'm not sure that includes `.write` but you should be able to use `stdout.write` instead.

Reply:
I'll respond myself: 

- with the `then-catch` approach, some logic can be applied even before the actual runtime(?) resolution of the promise. Maybe better in some more complex code structure where errors that are promise-related (interactions with other systems?) need to be addressed preemptively.
- the `awaits` keyword should be preferred for simpler code structure (not nested) and readability